src/features/profile/components/ProfileHeader.tsx

```diff
@@ -36,12 +37,13 @@ const getDisplayedExpirationDate = (expirationDate: Date, isUnderageBeneficiary:
 export function ProfileHeader(props: ProfileHeaderProps) {
   const { user } = props
   const isUnderageBeneficiary = isUserUnderageBeneficiary(user)
+  const { isLoggedIn } = useAuthContext()
 
-  if (!user) {
+  if (!isLoggedIn) {
```

Comment:
En dessous, `user` doit être défini, ce qui t'a obligé à mettre des `?` de ne pas avoir conserver cette condition, je te propose

if (!isLoggedIn || !user)` avec un commentaire du style `// tu not use ? bellow we also test user here`